### PR TITLE
Test for display_name in 3pid m.room.member events

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: tests/30rooms/12thirdpartyinvite.pl [hunk 1]
```diff
@@ -89,6 +89,17 @@ sub can_invite_unbound_3pid
    })->then( sub {
       $id_server->bind_identity( $hs_uribase, "email", $invitee_email, $invitee );
    })->then( sub {
+      matrix_get_room_state( $inviter, $room_id,
+         type      => "m.room.member",
+         state_key => $invitee->user_id,
+      )
+   })->then( sub {
+      my ( $body ) = @_;
+      log_if_fail "Body", $body;
```
Comment (leonerd) [inline tests/30rooms/12thirdpartyinvite.pl]: "Body" is a little vague - "Room state" maybe?

File: tests/30rooms/12thirdpartyinvite.pl [hunk 2]
```diff
@@ -89,6 +89,17 @@ sub can_invite_unbound_3pid
...
+      $body->{third_party_invite}{display_name} eq "Bob" or
```
Comment (leonerd) [inline tests/30rooms/12thirdpartyinvite.pl]: assert_eq( $body->{third_party_invite}{display_name}, "Bob",
       'invite display_name' );

File: tests/30rooms/12thirdpartyinvite.pl (hunk 1 above)
Comment (illicitonion) [inline tests/30rooms/12thirdpartyinvite.pl]: Done

File: tests/30rooms/12thirdpartyinvite.pl [hunk 3]
```diff
@@ -89,6 +89,17 @@ sub can_invite_unbound_3pid
...
+         die "Expected third_party_invite display_name to be 'Bob'";
+      Future->done( 1 );
+   })->then( sub {
```
Comment (leonerd) [inline tests/30rooms/12thirdpartyinvite.pl]: The entire construction of

       Future->done(1);
    })->then( sub {

is effectively redundant. You can safely delete these two lines; though I'd suggest replacing them with an empty one to keep the separation between the two logical phases.

Comment (illicitonion) [inline tests/30rooms/12thirdpartyinvite.pl]: Done

File: tests/30rooms/12thirdpartyinvite.pl (hunk 2 above)
Comment (illicitonion) [inline tests/30rooms/12thirdpartyinvite.pl]: Done

**Pull request merged**